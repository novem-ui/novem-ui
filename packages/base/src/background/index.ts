import { css } from '@emotion/react'
import { BackgroundProps } from '../types'

const background = ({
  background,
  backgroundAttachment,
  backgroundBlendMode,
  backgroundClip,
  backgroundColor,
  backgroundImage,
  backgroundOrigin,
  backgroundPosition,
  backgroundPositionX,
  backgroundPositionY,
  backgroundRepeat,
  backgroundSize
}: BackgroundProps) => css`
  ${background && `background: ${background};`}
  ${backgroundAttachment && `background-attachment: ${backgroundAttachment};`}
  ${backgroundAttachment && `background-blend-mode: ${backgroundBlendMode};`}
  ${backgroundClip && `background-clip: ${backgroundClip};`}
  ${backgroundColor && `background-color: ${backgroundColor};`}
  ${backgroundImage && `background-image: ${backgroundImage};`}
  ${backgroundOrigin && `background-origin: ${backgroundOrigin};`}
  ${backgroundPosition && `background-position: ${backgroundPosition};`}
  ${backgroundPositionX && `background-position-x: ${backgroundPositionX};`}
  ${backgroundPositionY && `background-position-y: ${backgroundPositionY};`}
  ${backgroundRepeat && `background-repeat: ${backgroundRepeat};`}
  ${backgroundSize && `background-size: ${backgroundSize};`}
`
export default background
