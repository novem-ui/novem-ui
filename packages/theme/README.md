# `@novem-ui/theme`

> TODO: description

## Usage

```
const theme = require('@novem-ui/theme');

// TODO: DEMONSTRATE API
```
