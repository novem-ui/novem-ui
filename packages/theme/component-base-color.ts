export { ComponentBaseColorProps, ComponentBaseColorProvider, useComponentBaseColor } from './src/component-base-color'
