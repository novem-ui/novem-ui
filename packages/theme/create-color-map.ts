export { default } from './src/utils/create-color-map'
