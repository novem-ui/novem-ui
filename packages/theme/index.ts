export { default as useTheme } from './use-theme'
export { default as withTheme } from './with-theme'
export { default as createColorMap } from './create-color-map'
export { default as ThemeProvider } from './theme-provider'
export { default, Theme, BaseColor, Color, HeadingSizeValue, SpacingValue, TextSizeValue, Shadows } from './theme'
