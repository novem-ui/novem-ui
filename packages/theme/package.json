{
  "name": "@novem-ui/theme",
  "version": "0.0.2-alpha.0",
  "description": "Theme for novem-ui ",
  "author": "Leonel Vieyra <me@leonelv.com>",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/theme#readme",
  "license": "ISC",
  "main": "index.js",
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/theme"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/theme --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.1.5",
    "polished": "^4.1.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
