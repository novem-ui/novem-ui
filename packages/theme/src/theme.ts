import colors from './colors'
import * as typography from './typography'
import spacing from './spacing'
import { WeightedColorKey } from '../types'

const theme = {
  colors,
  typography,
  spacing,
}

export type BaseColor = `${keyof typeof colors.palette}`
export type Color = `${keyof typeof colors.palette}.${WeightedColorKey}`
export type HeadingSizeValue = keyof typeof typography.headingFontSizes
export type SpacingValue = keyof typeof spacing
export type TextSizeValue = keyof typeof typography.textFontSizes
export type Theme = typeof theme


export default theme
