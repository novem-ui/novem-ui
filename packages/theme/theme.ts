export { default, Theme, BaseColor, Color, HeadingSizeValue, SpacingValue, TextSizeValue } from './src/theme'
