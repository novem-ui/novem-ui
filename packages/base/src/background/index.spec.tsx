import React from 'react'

import { render } from '@testing-library/react'
import styled from '@emotion/styled'

import theme from '@novem-ui/theme'
import background from '.'
import type { BackgroundProps } from '../types'

const ComponentWithBackgroundProps = styled.div<BackgroundProps>`
  ${background}
`

describe('layout', () => {
  test('should create a css fragment with the mapped props (partial)', () => {
    const { container } = render(<ComponentWithBackgroundProps theme={theme} backgroundClip="revert" />)

    expect(container).toMatchSnapshot()
  })

  test('should create a css fragment with the mapped props', () => {
    const props: BackgroundProps = {
      background: 'ThreeDFace',
      backgroundAttachment: 'initial',
      backgroundBlendMode: 'soft-light',
      backgroundClip: 'revert',
      backgroundColor: 'red.800',
      backgroundImage: 'url(image)',
      backgroundOrigin: 'inherit',
      backgroundPosition: 'right',
      backgroundPositionX: 'x-start',
      backgroundPositionY: 'y-end',
      backgroundRepeat: 'repeat-y',
      backgroundSize: 'unset'
    }
    const { container } = render(<ComponentWithBackgroundProps theme={theme} {...props} />)

    expect(container).toMatchSnapshot()
  })
})
