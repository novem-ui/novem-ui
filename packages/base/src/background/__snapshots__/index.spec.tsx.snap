// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout should create a css fragment with the mapped props (partial) 1`] = `
.emotion-0 {
  -webkit-background-clip: revert;
  background-clip: revert;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`layout should create a css fragment with the mapped props 1`] = `
.emotion-0 {
  background: ThreeDFace;
  background-attachment: initial;
  background-blend-mode: soft-light;
  -webkit-background-clip: revert;
  background-clip: revert;
  background-color: #9f1313;
  background-image: url(image);
  background-origin: inherit;
  -webkit-background-position: right;
  background-position: right;
  background-position-x: x-start;
  background-position-y: y-end;
  background-repeat: repeat-y;
  -webkit-background-size: unset;
  background-size: unset;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;
